recipes/android-ndk/all/conandata.yml

```diff
@@ -9,10 +19,6 @@ sources:
         "x86_64":
           url: "https://dl.google.com/android/repository/android-ndk-r23-linux.zip"
           sha256: "e3eacf80016b91d4cd2c8ca9f34eebd32df912bb799c859cc5450b6b19277b4f"
-      "Macos":
-        "x86_64":
-          url: "https://dl.google.com/android/repository/android-ndk-r23-darwin.zip"
```

Comment:
Thanks for taking a look. As I state in my PR description, the URL is still valid, but in my tests, I believe the package to be broken as-is for r23 on macOS. I don't believe the compilers are properly setup due to the zip file containing Mac Aliases, and the python copy function ends up making those files non-functional. I can't figure out how to properly make them work, and I figured rather than leaving the package as-is, pretending that it works, might as well officially make it un-supported.
If you want more details, and maybe want to take a look at it with me, I'd love to connect with you on a call at some point to discuss it further.

Reply:
what are these `Mac Aliases`? symbolic links or something else?

Reply:
Unfortunately, they're a special file type that Apple invented.
Here are some links about them:
https://www.cnet.com/tech/computing/os-x-in-depth-aliases-vs-symbolic-links/
https://en.wikipedia.org/wiki/Alias_(Mac_OS)

Essentially, the files are stored correctly in the downloaded zip files. However, we perform a self.copy() on those files, and they somehow turn into normal files. Now, the architecture-specific compiler files inside the (macOS) NDK are actually just script files that run the `clang` alias with arguments. But because the `clang` file (which was an alias) is now just a normal file, the commands fail. Further, just `chmod`ing the `clang` file doesn't help. So you go from a PERMISSION_DENIED error (trying to execute the normalized alias file) to an error about it not knowing how to execute the file that the alias used to point to. 

I'd recommend loading up the NDK r23 on macOS and taking a look. But as far as I can tell, this package is currently broken for macOS, and I'm not sure how to convert the file back into an alias (or prevent it from getting its alias property stripped).

Reply:
okay, I'll take a look on Mac today and let you know. maybe it's an easy fix. I've heard about these resource forks before, but rarely encountered them in practice.

Reply:
Just some more details for you.
Here's what I see when looking at the NDK directly:
```
jmcbride@Justins-MBP ~/working/ndk
$ ls -l android-ndk-r23/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++
lrwxr-xr-x@ 1 jmcbride  staff  5 Jul 30 18:07 android-ndk-r23/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ -> clang

jmcbride@Justins-MBP ~/working/ndk  [11s 00:05:07]
$ ./android-ndk-r23/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++
clang-12: error: no input files
```

And here it is from the perspective of the package directory:
```
jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin  [0s 00:13:25]
$ ./clang
-bash: ./clang: Permission denied

jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin  [0s 00:13:32]
$ ls -l ./clang
-rw-r--r--  1 jmcbride  staff  8 Nov 24 00:11 ./clang

jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin  [0s 00:13:38]
$ chmod +x ./clang

jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin
$ cat ./clang
clang-12

jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin  [0s 00:13:56]
$ chmod +x ./clang

jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin  [0s 00:14:08]
$ ./clang
./clang: line 1: clang-12: command not found

jmcbride@Justins-MBP ~/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/bin  [0s 00:14:11]
$ ls -l ./clang-12
-rwxr-xr-x  1 jmcbride  staff  145422112 Nov 24 00:11 ./clang-12
```

Reply:
I've checked archives of r23 and r23b, and I didn't find any resource forks or aliases within `toolchains/llvm/prebuilt/darwin-x86_64/bin`. it seems to be just a permissions issue. continue looking.

Reply:
okay, I've tried to build r23b locally and reproduced your problem
you have hit an already known problem with unzipping symbolic links (https://github.com/conan-io/conan/issues/9469)
it has nothing to do with macOS aliases, but I agree it's a bit obscure to debug
so the simple fix you want to do:
```
diff --git a/recipes/android-ndk/all/conanfile.py b/recipes/android-ndk/all/conanfile.py
index 4c13541eb..0e6cd2674 100644
--- a/recipes/android-ndk/all/conanfile.py
+++ b/recipes/android-ndk/all/conanfile.py
@@ -36,7 +36,7 @@ class AndroidNDKConan(ConanFile):
             raise ConanInvalidConfiguration(f"os,arch={self.settings.os},{self.settings.arch} is not supported by {self.name} (no binaries are available)")

     def build(self):
-        if self.version in ['r23']:
+        if self.version in ['r23', 'r23b']:
             data = self.conan_data["sources"][self.version]["url"][str(self.settings.os)][str(self.settings.arch)]
             unzip_fix_symlinks(url=data["url"], target_folder=self._source_subfolder, sha256=data["sha256"])
         else:
```
in order to work around the issue mentioned with symbolic links in zip archives

but then I have found that there are some broken symlinks in r23b itself (they are broken in downloaded .zip, not just in conan package), e.g. this one: `android-ndk-r23b/toolchains/llvm/prebuilt/darwin-x86_64/i686-linux-android/bin/as`

but about a previous version `r23` - there is nothing broken, so please keep it

Reply:
OK, I see what you mean. I'm not sure how I got so far off the beaten path with that investigation. I'll put the r23 macOS back. What do you think about r23b though? Should I just exclude it, and only add r23b Windows and Linux?
Thanks for the help. Much appreciated.

Reply:
it seems that `r23b` macOS distributioon is broken upstream, as there are broken symlinks. can you double-check on your side?
we can skip broken symlinks from packaging, but if someone relies on them, bad luck.
I suggest you checking it and reporting an issue upstream, if possible.
for conan package, you may skip `r23b` for macOS, and only use Windows/Linux for now.

Reply:
UPD: it also might be worth checking this issue: https://github.com/android/ndk/issues/1569, it might contain some relevent information

Reply:
Ah yeah, so it seems like a known issue. Good find. I browsed the issue list, but neglected to read through all the way to the bottom, which is where they confirm that the symlinks are broken. I also verified it locally on my side:
```
jmcbride@Justins-MBP ~/working/ndk/android-ndk-r23b  [0s 10:42:50]
$ find -L . -type l -ls
9650111        0 lrwxr-xr-x    1 jmcbride         staff                 149 Sep 29 16:54 ./toolchains/llvm/prebuilt/darwin-x86_64/aarch64-linux-android/bin/as -> /Volumes/Android/buildbot/src/android/ndk-release-r23/out/darwin/android-ndk-r23b/toolchains/llvm/prebuilt/darwin-x86_64/bin/aarch64-linux-android-as
9652181        0 lrwxr-xr-x    1 jmcbride         staff                 149 Sep 29 16:54 ./toolchains/llvm/prebuilt/darwin-x86_64/arm-linux-androideabi/bin/as -> /Volumes/Android/buildbot/src/android/ndk-release-r23/out/darwin/android-ndk-r23b/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-as
9652186        0 lrwxr-xr-x    1 jmcbride         staff                 148 Sep 29 16:54 ./toolchains/llvm/prebuilt/darwin-x86_64/x86_64-linux-android/bin/as -> /Volumes/Android/buildbot/src/android/ndk-release-r23/out/darwin/android-ndk-r23b/toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-as
9652191        0 lrwxr-xr-x    1 jmcbride         staff                 146 Sep 29 16:54 ./toolchains/llvm/prebuilt/darwin-x86_64/i686-linux-android/bin/as -> /Volumes/Android/buildbot/src/android/ndk-release-r23/out/darwin/android-ndk-r23b/toolchains/llvm/prebuilt/darwin-x86_64/bin/i686-linux-android-as
```
Looks like they're targeting a fix for r23c. I tried wrapping the `open` call in a `try/catch` to catch the `FileNotFoundError`, but unfortunately a higher level seems to then complain: 
```
ERROR: The file is a broken symlink, verify that you are packaging the needed destination files: '/Users/jmcbride/.conan/data/android-ndk/r23b/_/_/package/46f53f156846659bf39ad6675fa0ee8156e859fe/toolchains/llvm/prebuilt/darwin-x86_64/aarch64-linux-android/bin/as'.You can skip this check adjusting the 'general.skip_broken_symlinks_check' at the conan.conf file.
```
Ultimately decided I'm a bit out of my depth here and I'll leave it to someone else by simply excluding r23 Mac as you advised.

Reply:
Oh darn, it looks like this is ALSO true for the linux package. Due to not calling the `fix_symlinks` function for r23b initially (before your suggestion), that issue was masked when I was testing linux. I have removed the linux package as well.. What a nightmare of a package.

Reply:
well, it seems like this time it's broken upstream, so maybe wait for `r32c` or whatever else comes next, to re-add Linux/macOS packages

Reply:
also, if you (or someone else) have time and will, it's possible to try to repare these broken symlinks in the recipe as stated in comment: https://github.com/android/ndk/issues/1569#issuecomment-950155499
(not necessarily in this PR, it could be done later, or never).